core/src/actors/session/handlers.rs

```diff
@@ -542,16 +555,32 @@ fn send_item_msg(session: &mut Session, ctx: &mut Context<Session>, hash: &Hash)
     });
 }
 
-fn send_block_msg(session: &mut Session, block: &Block) {
-    let block_header = block.block_header.clone();
-    let proof = block.proof.clone();
-    let txns = block.txns.clone();
+fn send_block_msg(session: &mut Session, block: Block) {
+    let block_header = block.block_header;
+    let proof = block.proof;
+    let txns = block.txns;
     // Build Block msg
     let block_msg = WitnetMessage::build_block(block_header, proof, txns);
     // Send Block msg
     session.send_message(block_msg);
 }
 
+fn request_block_msg(session: &mut Session, block_entry: InventoryEntry) {
+    let mut inv_entries: Vec<InventoryEntry> = Vec::new();
```

Comment:
These two lines can be blended into a single one:
```suggestion
    // Initialize a new inventory entries vector with the given block entry as its sole member
    let mut inv_entries: Vec<InventoryEntry> = vec![block_entry];
```